api/kessel/inventory/v1beta2/resource.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+package kessel.inventory.v1beta2;
+import "kessel/inventory/v1beta2/reporter_data.proto";
+import "google/protobuf/struct.proto";
+import "buf/validate/validate.proto";
+
+option go_package = "github.com/project-kessel/inventory-api/api/kessel/inventory/v1beta2";
+option java_multiple_files = true;
+option java_package = "org.project_kessel.api.inventory.v1beta2";
+
+message Resource {
+
+ string resource_type = 1 [(buf.validate.field).string = {min_len: 1}];
+ ReporterData reporter_data = 2 [(buf.validate.field).required = true];
+ google.protobuf.Struct common_resource_data = 3 [(buf.validate.field).required = true, json_name = "commonResourceData"];
+
```

Comment:
Curious, what can a client do to create a proto struct, given the JSON schema, in a tool-assisted way?

e.g. I can use JSON schema to generate types (e.g in python or go or whatever) which I can use to create valid objects adhering that schema. What's it like to go from that object, to the gRPC API, where it requires an equivalent proto struct?

It would be interested to see if this is something the client libraries could help handle, though that needn't be a requirement for now.

Reply:
Added a Jira task for this https://issues.redhat.com/browse/RHCLOUD-39149